content/blog/2016/2016-12-13-jenkins-puppet-enterprise-plugin.adoc

```diff
@@ -0,0 +1,271 @@
+update nodes across a set of nodes that match certain criteria. In each
+scenario, Puppet will always push distributed changes in the correct order by
+respecting the cross-node dependencies.
+
+To create a job in the Puppet orchestrator from a Jenkins pipeline, use the
+puppet.job method. The puppet.job method will create a new orchestrator job,
+monitor the job for completion, and determine if any Puppet runs failed. If
+there were failures, the pipeline will fail.
+
+To run Puppet against all of production:
+
+----
+puppet.job 'production'
+----
+
+To run Puppet against instances of an application in production:
+
+----
+puppet.job 'production', application: 'Myapp'
+----
+
+To run Puppet against nodes db.example.com, appserver01.example.com, and appserver02.example.com:
+
+----
+puppet.job 'production', nodes: ['db.example.com','appserver01.example.com','appserver02.example.com']
+----
+
+To run Puppet against all Red Hat nodes in the AWS us-west-2c region that were
+created in the last 24 hours using a PQL query:
+
+----
+puppet.job 'production', query: 'inventory[certname] { facts.os.name = "RedHat" and facts.ec2_metadata.placement.availability-zone = "us-west-2c" and uptime_hours < 24 }'
+----
+
+As you can see, the puppet.job command means you can be as broad or as targeted
+as you need to be for different parts of your pipeline. There are many other
+options you can add to the puppet.job method call, such as setting the Puppet
+runs to noop, or giving the orchestrator a maximum concurrency limit.
+link:https://puppet.com/product/capabilities/application-orchestration[Learn
+more about the orchestrator here.]
+
+== Updating Puppet code
+
+If you're using code management in Puppet Enterprise (and you should), you can
+ensure that all the modules, site manifests, Hiera data, and roles and profiles
+are staged, synced, and ready across all your Puppet masters, direct from your
+Jenkins pipeline.
+
+To update Puppet code across all Puppet masters, use the puppet.codeDeploy method.
+
+----
+puppet.codeDeploy 'staging'
+----
+
+link:https://puppet.com/product/capabilities/code-management[Learn more code management in Puppet Enterprise here.]
+
+== Setting Hiera values
+
+The plugin includes an experimental feature to set Hiera key/value pairs. There
+are many cases where you need to promote information through a pipeline, such
+as a build version or artifact location. Doing so is very difficult in Puppet,
+since data promotion almost always involves changing Hiera files and committing
+to version control.
+
+The plugin exposes an HTTP API endpoint that Hiera can query using the
+hiera-http backend. With the backend configured on the Puppet master(s),
+key/value pairs can be set to scopes. A scope is arbitrary and can be anything
+you like, such as a Puppet environment, a node's certname, or the name of a
+Facter fact like operatingsystem or domain.
+
+To set a Hiera value from a pipeline, use the puppet.hiera method.
```

Comment:
A lot of technical terms (commands/methods/…) like puppet.hiera in this blog post that I'd print `monospace`. Not sure what the styleguide is on this though.